pkg/runtime/runtime.go

```diff
@@ -556,17 +558,19 @@ func (a *DaprRuntime) appHealthReadyInit(ctx context.Context) (err error) {
 		if err != nil {
 			log.Warn(err)
 		} else {
-			// Workflow engine depends on actor runtime being initialized
-			// This needs to be called before "SetActorsInitDone" on the universal API object to prevent a race condition in workflow methods
-			a.initWorkflowEngine(ctx)
-
 			a.daprUniversalAPI.SetActorRuntime(a.actor)
 		}
 	} else {
 		// If actors are not enabled, still invoke SetActorRuntime on the workflow engine with `nil` to unblock startup
-		a.workflowEngine.SetActorRuntime(nil)
+		err = setActorRuntimeForBackend(ctx, a.workflowEngine.Backend, nil)
+		if err != nil {
+			log.Errorf("error while setting actor runtime for backend to nil: %v", err)
```

Comment:
Shouldn't we be returning an error here instead of logging?
```suggestion
			return fmt.Errorf("error while setting actor runtime for backend to nil: %w", err)
```